agenta-cli/agenta/sdk/agenta_tracing.py

```diff
@@ -0,0 +1,106 @@
+# Stdlib Imports
+import uuid
+from datetime import datetime
+from typing import Optional, Dict, Any, Optional
+
+# Own Imports
+from agenta.client.backend import client
+from agenta.client.backend.types.span_status import SpanStatus
+from agenta.client.backend.client import AsyncObservabilityClient
+
+
+class LLMTracing:
+    def __init__(self, base_url: str, api_key: Optional[str] = None):
+        self.base_url = base_url + "/api"
+        self.api_key = api_key if api_key is not None else ""
+
+    def initialize_client(self) -> AsyncObservabilityClient:
+        return client.AsyncAgentaApi(
+            base_url=self.base_url, api_key=self.api_key, timeout=120  # type: ignore
+        ).observability
+
+    async def create_trace(
+        self,
+        client: AsyncObservabilityClient,
+        app_id: str,
+        base_id: str,
+        config_name: str,
+        **kwargs: Dict[str, Any],
+    ):
+        trace = await client.create_trace(
+            app_id=app_id,
+            base_id=base_id,
+            config_name=config_name,
+            cost=kwargs["cost"],  # type: ignore
+            environment=kwargs["environment"],  # type: ignore
+            status="INITIATED",
+            token_consumption=kwargs["total_tokens"],  # type: ignore
+            tags=[],
+        )
+        return trace
+
+    async def finalize_trace(
+        self, client: AsyncObservabilityClient, trace_id: str, status: str
+    ) -> bool:
+        return await client.update_trace(
+            trace_id=trace_id, status=status, end_time=datetime.now()
+        )
+
+    async def create_span(
```

Comment:
how would the user access this function? How can they create a span in the call?

Reply:
They won't actually. The tracing is designed for the SDK to access the function. I will document my thoughts and during our upcoming sync meeting, I will explain how it would work for you to provide your feedback. Thank you.

Reply:
I wrote some thoughts in the document about how I think it should work